protos/sigstore_trustroot.proto

```diff
@@ -117,3 +117,43 @@ message TrustedRoot {
         // A set of trusted timestamping authorities.
         repeated CertificateAuthority timestamp_authorities = 5;
 }
+
+
+// SigningConfig represents the trusted entities/state needed by Sigstore
+// signing. In particular, it primarily contains service URLs that a Sigstore
+// signer may need to connect to for the online aspects of signing.
+message SigningConfig {
+        // A URL to a Fulcio-compatible CA, capable of receiving
+        // Certificate Signing Requests (CSRs) and responding with
+        // issued certificates.
+        //
+        // This URL **MUST** be the "base" URL for the CA, which clients
+        // should construct an appropriate CSR endpoint on top of.
+        // For example, if `fulcio_url` is `https://example.com/ca`, then
+        // the client **MAY** construct the CSR endpoint as
+        // `https://example.com/ca/api/v2/signingCert`.
+        string fulcio_url = 1;
```

Comment:
I was originally going to call this `csr_url` (and similar `tlog_url` for Rekor below), but in practice the APIs here are Fulcio- and Rekor-specific. Curious if anybody has strong thoughts here.

Reply:
I think it would be more consistent if they matched the types above?

Reply:
I agree, although that _might_ be confusing: technically verification requires none of the machinery specific to Rekor or Fulcio (so we can speak in general CA, CT, etc. terms), while signing _concretely_ relies on REST APIs exposed by Fulcio or Rekor (or services imitating those).

Then again, maybe that can be solved by just documenting in the comments that these URLs are expected to adhere to the Fulcio/Rekor APIs. If that seems reasonable to you, I'll rename and update for that 🙂

Reply:
Yeah seems reasonable.

Reply:
PTAL!